libraries/AP_Compass/AP_Compass_AK8963.cpp

```diff
@@ -226,6 +232,11 @@ bool AP_Compass_AK8963_MPU9250::init()
     if (_backend == NULL) {
         hal.scheduler->panic(PSTR("_backend coudln't be allocated"));
     }
+    if (!_backend->init()) {
```

Comment:
Shouldn't this be an else if so that we don't access a member function of a null class?